File: src/qvoid/paths.py
from __future__ import annotations

import os
from pathlib import Path


def config_dir() -> Path:
    xdg = os.environ.get("XDG_CONFIG_HOME")
    base = Path(xdg) if xdg else Path.home() / ".config"
    return base / "qvoid"


def data_dir() -> Path:
    xdg = os.environ.get("XDG_DATA_HOME")
    base = Path(xdg) if xdg else Path.home() / ".local" / "share"
    return base / "qvoid"


def registry_path() -> Path:
    return config_dir() / "collections.toml"


def collection_config_path(name: str) -> Path:
    return config_dir() / "collections" / f"{name}.toml"


def collection_data_dir(name: str) -> Path:
    return data_dir() / name

File: src/paths.ts
import { homedir } from "os";
import { join } from "path";

function configBase(): string {
  return process.env["XDG_CONFIG_HOME"] ?? join(homedir(), ".config");
}

function dataBase(): string {
  return process.env["XDG_DATA_HOME"] ?? join(homedir(), ".local", "share");
}

export function configDir(): string {
  return join(configBase(), "qvoid");
}

export function dataDir(): string {
  return join(dataBase(), "qvoid");
}

export function registryPath(): string {
  return join(configDir(), "collections.toml");
}

export function collectionConfigPath(name: string): string {
  return join(configDir(), "collections", `${name}.toml`);
}

export function collectionDataDir(name: string): string {
  return join(dataDir(), name);
}

// ML classifier model/training data is shared across all collections, so it lives
// directly under the data dir rather than under a per-collection subdirectory.
export function mlModelPath(): string {
  return join(dataDir(), "classifier.json");
}

export function mlTrainingDataPath(): string {
  return join(dataDir(), "training_data.json");
}
